libvast/include/vast/query_context.hpp

```diff
@@ -123,6 +123,9 @@ struct query_context {
   /// The query expression.
   expression expr = {};
 
+  /// The initial taste size.
+  std::optional<uint32_t> taste = std::nullopt;
```

Comment:
Consider using just an uint32_t with a default value

Reply:
Nvm, the default value is not known at compile time because we have the `vast.max-taste-partitions` config setting